src/wkt/tests/message_with_i32.rs

```diff
@@ -0,0 +1,272 @@
+        let want = MessageWithI32 {
+            optional: None,
+            ..Default::default()
+        };
+        let got = serde_json::from_value::<MessageWithI32>(input)?;
+        assert_eq!(got, want);
+        Ok(())
+    }
+
+    #[test_case("123", 123)]
+    #[test_case(456, 456)]
+    #[test_case("-789", -789)]
+    fn test_repeated<T>(input: T, want: i32) -> Result
+    where
+        T: serde::ser::Serialize,
+    {
+        let value = json!({"repeated": [input]});
+        let got = serde_json::from_value::<MessageWithI32>(value)?;
+        let output = json!({"repeated": [want]});
+        if want == 0 {
+            return Ok(());
+        }
+        assert_eq!(
+            got,
+            MessageWithI32 {
+                repeated: vec![want],
+                ..Default::default()
+            }
+        );
+        let trip = serde_json::to_value(&got)?;
+        assert_eq!(trip, output);
+        Ok(())
+    }
+
+    #[test_case(json!({"repeated": []}))]
+    #[test_case(json!({}))]
+    fn test_repeated_default(input: Value) -> Result {
+        let want = MessageWithI32 {
+            repeated: vec![],
+            ..Default::default()
+        };
+        let got = serde_json::from_value::<MessageWithI32>(input)?;
+        assert_eq!(got, want);
+        let output = serde_json::to_value(&got)?;
+        assert_eq!(output, json!({}));
+        Ok(())
+    }
+
+    #[test_case(0, 0)]
+    #[test_case("0", 0; "zero string")]
+    #[test_case("123", 123)]
+    #[test_case(456, 456)]
+    #[test_case("-789", -789)]
+    fn test_map_value<T>(input: T, want: i32) -> Result
+    where
+        T: serde::ser::Serialize,
+    {
+        let value = json!({"mapValue": {"test": input}});
+        let got = serde_json::from_value::<MessageWithI32>(value)?;
+        let output = json!({"mapValue": {"test": want}});
+        assert_eq!(
+            got,
+            MessageWithI32 {
+                map_value: HashMap::from([("test".to_string(), want)]),
+                ..Default::default()
+            }
+        );
+        let trip = serde_json::to_value(&got)?;
+        assert_eq!(trip, output);
+        Ok(())
+    }
+
+    #[test_case(json!({"mapValue": {}}))]
+    #[test_case(json!({}))]
+    fn test_map_value_default(input: Value) -> Result {
+        let want = MessageWithI32::default();
+        let got = serde_json::from_value::<MessageWithI32>(input)?;
+        assert_eq!(got, want);
+        let output = serde_json::to_value(&got)?;
+        assert_eq!(output, json!({}));
+        Ok(())
+    }
+
+    #[test_case("0", 0)]
+    #[test_case("123", 123)]
+    #[test_case("-789", -789)]
+    fn test_map_key<T>(input: T, want: i32) -> Result
+    where
+        T: Into<String>,
+    {
+        let value = json!({"mapKey": {input: "test"}});
+        let got = serde_json::from_value::<MessageWithI32>(value)?;
+        let output = json!({"mapKey": {want.to_string(): "test"}});
```

Comment:
`to_string()`? shouldn't we reserialize these without quotes?

(I might be lost here)